src/request.rs

```diff
@@ -113,17 +117,24 @@ impl DispatchSignedRequest for Client {
                 debug!("{}:{}", h.name(), h.value_string());
             }
         }
-
         let mut hyper_response = match request.payload() {
             None => try!(self.request(hyper_method, &final_uri).headers(hyper_headers).body("").send()),
             Some(payload_contents) => try!(self.request(hyper_method, &final_uri).headers(hyper_headers).body(payload_contents).send()),
         };
-
         let mut body = String::new();
-        try!(hyper_response.read_to_string(&mut body));
+        let mut body_as_bytes : Vec<u8> = Vec::new();
+        let response_is_binary = match hyper_response.headers.get::<ContentType>() {
```

Comment:
Might be a little cleaner here to pull this out into an `is_binary()` function, so instead of the `match` on true/false below you could just say `if is_binary(hyper_response)`.  Just nitpicking though.